### ISSUE TITLE
could not be resolved to an NgModule class. - Error in compile with Ivy - Angular 9

### ISSUE THREAD

**Issue by abcfoundry** [nstudio/nativescript-checkbox#129]
Trying to install and use the plugin getting error.  See below.

ERROR in node_modules/@nstudio/nativescript-checkbox/angular/index.d.ts:14:22 - error NG6002: Appears in the NgModule.imports of DesignFormsModule, but could not be resolved to an NgModule class.

This likely means that the library (@nstudio/nativescript-checkbox/angular) which declares TNSCheckBoxModule has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.

14 export declare class TNSCheckBoxModule {
                        ~~~~~~~~~~~~~~~~~

Webpack compilation complete. Watching for file changes.
Webpack build done!
Project successfully prepared (ios)
Successfully transferred all files on device 4150A0D5-345D-4DE1-93B8-1B13742C6687.
Restarting application on device 4150A0D5-345D-4DE1-93B8-1B13742C6687...
(RunningBoardServices) [com.apple.runningboard:connection] Identity resolved as application<org.nativescript.designsystem>





package.json

`
{
  "name": "",
  "license": "SEE LICENSE IN <your-license-filename>",
  "version": "0.0.0",
  "nativescript": {
    "id": "",
    "tns-android": {
      "version": "6.5.0"
    },
    "tns-ios": {
      "version": "6.5.0"
    }
  },
  "scripts": {
    "reset": "npx rimraf -- hooks node_modules platforms && npm run js.clean",
    "clean": "npm run js.clean",
    "js.clean": "npx rimraf -- app/**/*.js",
    "prepare.ios.release": "tns prepare ios --release",
    "prepare.android.release": "tns prepare android --release",
    "build.ios": "tns build ios --env.aot --env.uglify --env.production --release",
    "build.android": "tns build android --env.aot --env.uglify --env.production --release",
    "debug.ios": "tns debug ios",
    "debug.android": "tns debug android",
    "postinstall": "node ./tools/xplat-postinstall.js && npm run ngcc",
    "ngcc": "ngcc --properties es2015 module main --first-only"
  },
  "dependencies": {
    "@alpha2k/nativescript-scss": "file:../../xplat/nativescript/scss",
    "@alpha2k/scss": "file:../../libs/scss",
    "@angular/animations": "~9.1.0",
    "@angular/common": "~9.1.0",
    "@angular/compiler": "~9.1.0",
    "@angular/core": "~9.1.0",
    "@angular/forms": "~9.1.0",
    "@angular/platform-browser": "~9.1.0",
    "@angular/platform-browser-dynamic": "~9.1.0",
    "@angular/router": "~9.1.0",
    "@nativescript/angular": "https://github.com/nstudio/ns-ng/blob/master/nativescript-angular-9.0.0.tgz?raw=true",
    "@nativescript/theme": "^2.3.3",
    "@ngx-translate/core": "~12.1.0",
    "@nstudio/nativescript-floatingactionbutton": "^2.0.0",
    "@nstudio/nativescript-checkbox": "^1.0.0",
    "nativescript-angular": "https://github.com/nstudio/ns-ng/blob/master/nativescript-angular-compat.tgz?raw=true",
    "nativescript-intl": "~3.0.0",
    "nativescript-ngx-fonticon": "~6.0.3",
    "rxjs": "^6.5.0",
    "tns-core-modules": "~6.4.0",
    "tslib": "~1.11.0",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~9.1.0",
    "@angular/language-service": "~9.1.0",
    "@ngtools/webpack": "~9.1.0",
    "codelyzer": "~5.1.0",
    "nativescript-dev-webpack": "~1.5.0",
    "terser-webpack-plugin": "~2.2.0",
    "tns-platform-declarations": "~6.4.0",
    "typescript": "~3.7.4"
  }
}
`

**Comment by sublime392** [comment]
Even after adding this module to the ngcc.config.js, it does not compile. I believe it is because it is not the '@nativescript/angular' namespace in its imports:
import { BaseValueAccessor } from 'nativescript-angular/forms/value-accessors/base-value-accessor';

**Comment by abcfoundry** [comment]
Hey @sublime392 we have it working.  This is the process we followed thanks to @NathanWalker 

https://github.com/NativeScript/nativescript-angular/wiki/Updating-and-developing-for-@nativescript-angular-v9-with-Ivy-or-without 

    "@nstudio/nativescript-checkbox": {
      entryPoints: {
        "angular": {
          override: {
            main: "./index.js",
            typings: "./index.d.ts",
          },
          ignoreMissingDependencies: true,
        }
      },
      ignorableDeepImportMatchers: [
        /tns-core-modules\//,
        /@nativescript\/core\//,
        /@nativescript\/angular\//,
        /nativescript-angular\//,
        /nativescript-angular\/forms\//,
        /@angular\/core\//,
        /@angular\/forms\//,
      ]
    },

**Comment by madmas** [comment]
Hi @abcfoundry @sublime392 , using this config, I still get these errors: 
```
ERROR in ../node_modules/@nstudio/nativescript-checkbox/angular/index.js
Module not found: Error: Can't resolve 'nativescript-angular/element-registry' in '/app/node_modules/@nstudio/nativescript-checkbox/angular'
 @ ../node_modules/@nstudio/nativescript-checkbox/angular/index.js 5:25-73
 @ ./app/app.module.ts
 @ ./main.ts
ERROR in ../node_modules/@nstudio/nativescript-checkbox/angular/index.js
Module not found: Error: Can't resolve 'nativescript-angular/forms/value-accessors/base-value-accessor' in '/app/node_modules/@nstudio/nativescript-checkbox/angular'
 @ ../node_modules/@nstudio/nativescript-checkbox/angular/index.js 6:28-101
 @ ./app/app.module.ts
 @ ./main.ts
ERROR in ../node_modules/@nstudio/nativescript-checkbox/angular/index.js
Module not found: Error: Can't resolve 'nativescript-angular/lang-facade' in '/app/node_modules/@nstudio/nativescript-checkbox/angular'
 @ ../node_modules/@nstudio/nativescript-checkbox/angular/index.js 7:20-63
 @ ./app/app.module.ts
 @ ./main.ts
```

Do you have a hint for me what I can do about them?

**Comment by abcfoundry** [comment]
having a look now... will report back

**Comment by chirprooms** [comment]
Getting the same problem

Nativescript 7 with Angular 10 using the config above.

`ERROR in ../node_modules/@nstudio/nativescript-checkbox/__ivy_ngcc__/angular/index.js
Module not found: Error: Can't resolve '../checkbox' in '/Users/aedanmcgarry/HTDOCS/other/no-strings-app/node_modules/@nstudio/nativescript-checkbox/__ivy_ngcc__/angular'
 @ ../node_modules/@nstudio/nativescript-checkbox/__ivy_ngcc__/angular/index.js 22:49-71
 @ ./app/pages/profile/profile.module.ts
 @ ./app/app.routes.ts
 @ ./app/app-routing.module.ts
 @ ./app/app.module.ts
 @ ./main.ts

ERROR in ../node_modules/@nstudio/nativescript-checkbox/__ivy_ngcc__/angular/index.js
Module not found: Error: Can't resolve '@nativescript/angular/forms/value-accessors' in '/Users/aedanmcgarry/HTDOCS/other/no-strings-app/node_modules/@nstudio/nativescript-checkbox/__ivy_ngcc__/angular'
 @ ../node_modules/@nstudio/nativescript-checkbox/__ivy_ngcc__/angular/index.js 4:0-80 68:2-19
 @ ./app/pages/profile/profile.module.ts
 @ ./app/app.routes.ts
 @ ./app/app-routing.module.ts
 @ ./app/app.module.ts
 @ ./main.ts`

**Comment by harishchopra86** [comment]
Getting error in installing on native script 7.0.8 with angular 10.

ERROR in node_modules/@nstudio/nativescript-checkbox/angular/index.d.ts:14:22 - error NG6002: Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class.

This likely means that the library (@nstudio/nativescript-checkbox/angular) which declares TNSCheckBoxModule has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.

14 export declare class TNSCheckBoxModule {

**Comment by harishchopra86** [comment]
Hello Contributors(@bradmartin @triniwiz @EddyVerbruggen @sitefinitysteve @NathanWalker @manojdcoder @lochstar @jogboms @nstudio),
Guys, any update or advise on this issue.

**Comment by alereisan** [comment]
I am facing the same issue:

```
ERROR in node_modules/@nstudio/nativescript-checkbox/angular/index.d.ts:14:22 - error NG6002: Appears in the NgModule.imports of AuthModule, but could not be resolved to an NgModule class.

This likely means that the library (@nstudio/nativescript-checkbox/angular) which declares TNSCheckBoxModule has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.

14 export declare class TNSCheckBoxModule {
                        ~~~~~~~~~~~~~~~~~
node_modules/@nstudio/nativescript-checkbox/angular/index.d.ts:14:22 - error NG6002: Appears in the NgModule.imports of CheckoutModule, but could not be resolved to an NgModule class.

This likely means that the library (@nstudio/nativescript-checkbox/angular) which declares TNSCheckBoxModule has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.
```

package.json

```
{
  "name": "@nativescript/template-hello-world-ng",
  "main": "main.js",
  "version": "7.0.2",
  "author": "NativeScript Team <oss@nativescript.org>",
  "description": "NativeScript Application",
  "license": "SEE LICENSE IN <your-license-filename>",
  "publishConfig": {
    "access": "public"
  },
  "keywords": [
    "nativescript",
    "mobile",
    "angular",
    "{N}",
    "template"
  ],
  "repository": "<fill-your-repository-here>",
  "bugs": {
    "url": "https://github.com/NativeScript/NativeScript/issues"
  },
  "scripts": {
    "ngcc": "ngcc --properties es2015 module main --first-only",
    "postinstall": "npm run ngcc"
  },
  "dependencies": {
    "@angular/animations": "~10.1.0",
    "@angular/common": "~10.1.0",
    "@angular/compiler": "~10.1.0",
    "@angular/core": "~10.1.0",
    "@angular/forms": "~10.1.0",
    "@angular/platform-browser": "~10.1.0",
    "@angular/platform-browser-dynamic": "~10.1.0",
    "@angular/router": "~10.1.0",
    "@nativescript/angular": "~10.1.0",
    "@nativescript/core": "~7.0.0",
    "@nativescript/theme": "~2.3.0",
    "@ngx-translate/core": "^13.0.0",
    "@ngx-translate/http-loader": "^6.0.0",
    "@nstudio/nativescript-checkbox": "^2.0.2",
    "@nstudio/nativescript-snackbar": "^2.0.2",
    "@proplugins/nativescript-drop-down": "^9.0.0",
    "@proplugins/nativescript-pager": "^11.5.16",
    "moment": "^2.29.1",
    "moment-timezone": "^0.5.31",
    "nativescript-appversion": "^1.4.4",
    "nativescript-barcodescanner": "^4.0.1",
    "nativescript-bitmap-factory": "^1.8.1",
    "nativescript-brightness": "^1.0.1",
    "nativescript-email": "^1.6.0",
    "nativescript-google-maps-sdk": "^2.9.1",
    "nativescript-orientation": "^2.2.5",
    "nativescript-pager": "^12.0.0-alpha.9",
    "nativescript-phone": "^2.0.0",
    "nativescript-plugin-firebase": "^10.5.2",
    "nativescript-ui-listview": "^9.0.2",
    "reflect-metadata": "~0.1.12",
    "rxjs": "^6.6.0",
    "zone.js": "~0.11.1"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~10.1.0",
    "@nativescript/ios": "7.0.1",
    "@nativescript/types": "~7.0.0",
    "@nativescript/webpack": "~3.0.0",
    "@ngtools/webpack": "~10.1.0",
    "node-sass": "^4.14.1",
    "typescript": "~3.9.0"
  },
  "private": "true",
  "readme": "NativeScript Application"
}
```



Currently this is one of many issues when using Nativescript 7 + Angular 10, production app is not working on iOS14, so I am going to remove the checkbox plugin for now, and make a simple icon based checkbox component in angular instead.

**Comment by NathanWalker** [comment]
@alereisan @harishchopra86 @chirprooms This is resolved in `@nstudio/nativescript-checkbox` **2.0.3** published yesterday. Source is managed in plugin workspace here now which properly builds Angular 10.1.x support 👍 
https://github.com/nstudio/nativescript-plugins
